solutions/src/main/java/ru/sberbank/school/task13/cache/CacheProxy.java

```diff
@@ -0,0 +1,338 @@
+                return false;
+            }
+
+            if (method.getReturnType().equals(void.class)) {
+                throw new IllegalArgumentException("called method can not be void");
+            }
+
+            return true;
+        }
+
+        private Object invokeJVM(Method method, Object[] args) throws InvocationTargetException,
+                IllegalAccessException {
+
+            String keyFromInput = getKeyToCache(method, args);
+
+            Object result = findInCache(keyFromInput);
+
+            if (result == null) {
+                result = method.invoke(service, args);
+                cacheMap.put(keyFromInput, result);
+            }
+
+            return result;
+        }
+
+        private Object invokeFile(Method method, Object[] args)
+                throws InvocationTargetException, IllegalAccessException {
+
+            String key = getKeyToCache(method, args);
+            String fileName = key + ".cache";
+
+            File file = new File(directoryPath + File.separator + fileName);
+            Object result = null;
+            File fileInCache = findInDirectory(file);
+
+            if (fileInCache != null) {
+                result = loadFromCacheFile(fileInCache);
+            } else {
+                result = method.invoke(service, args);
+                writeToCacheFile(file, result);
+            }
+
+            return result;
+        }
+
+        private Object findInCache(String toFind) {
+            Object result = null;
+
+            if (cacheMap.containsKey(toFind)) {
+                result = cacheMap.get(toFind);
+            } else {
+                for (String keyFromCacheMap : cacheMap.keySet()) {
+                    if (keyFromCacheMap.contains(toFind)) {
+                        result = cacheMap.get(keyFromCacheMap);
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        private File findInDirectory(File toFind) {
+            String fullNameToFind = isZip(toFind.getAbsolutePath())
+                    ? (toFind.getAbsolutePath() + ".zip") : toFind.getAbsolutePath();
+
+            File result = null;
+            File toCheck = new File(fullNameToFind);
+
+            if (toCheck.exists() && ! toCheck.isDirectory()) {
+                result = new File(fullNameToFind);
+            }
+            return result;
+        }
+
+        private Object loadFromCacheFile(File fileToLoad) {
+            Object result = null;
+
+            if (isZip(fileToLoad.getName())) {
+                result = readZipFile(fileToLoad);
+            } else {
+                result = readFromFile(fileToLoad);
+            }
+
+            return result;
+        }
+
+        private Object readFromFile(File file) {
+            Object result = null;
+
+            try (FileInputStream fin = new FileInputStream(file);
+                 ObjectInputStream oin = new ObjectInputStream(fin)) {
+
+                    result = oin.readObject();
```

Comment:
&#39;try&#39; child has incorrect indentation level 20, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1244355778>*